hack/common.sh

```diff
@@ -1066,6 +1066,16 @@ function os::build::environment::withsource() {
 
   docker start "${container}" > /dev/null
   docker logs -f "${container}"
+
+  # extract content from the image
+  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
+    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
+    IFS=: eval 'for path in ${OS_BUILD_ENV_PRESERVE}; do
```

Comment:
I guess this would leak to the rest of the function after this call, so if you care about that case:
```sh
local OLD_IFS="${IFS}"
IFS=:
// for loop
export IFS="${OLD_IFS}"
```

In either case, I think staying away from `eval` is best since it's a gold mine of gotchas and hidden behavior.